extensions/netty-loom-adaptor/deployment/src/main/java/io/quarkus/netty/loom/adaptor/NettyLoomAdaptorProcessor.java

```diff
@@ -577,16 +595,310 @@ public MethodVisitor visitMethod(
+            mv.visitJumpInsn(IFNULL, L14);
+            mv.visitVarInsn(ALOAD, 10);
+            mv.visitVarInsn(ALOAD, 0);
+
+            mv.visitFieldInsn(GETFIELD, "io/netty/buffer/PooledByteBufAllocator", "trimTask",
+                    "Ljava/lang/Runnable;");
+            mv.visitFieldInsn(GETSTATIC, "io/netty/buffer/PooledByteBufAllocator",
+                    "DEFAULT_CACHE_TRIM_INTERVAL_MILLIS", "J");
+            mv.visitFieldInsn(GETSTATIC, "io/netty/buffer/PooledByteBufAllocator",
+                    "DEFAULT_CACHE_TRIM_INTERVAL_MILLIS", "J");
+            mv.visitFieldInsn(GETSTATIC, "io/netty/buffer/PooledByteBufAllocator", "MILLISECONDS",
+                    "Ljava/util/concurrent/TimeUnit;");
+            mv.visitMethodInsn(INVOKEINTERFACE, "io/netty/util/concurrent/EventExecutor",
+                    "scheduleAtFixedRate",
+                    "(Ljava/lang/Runnable;JJLjava/util/concurrent/TimeUnit;)Lio/netty/util/concurrent/ScheduledFuture;",
+                    true);
+            mv.visitInsn(POP);
+
+            mv.visitLabel(L14);
+            mv.visitVarInsn(ALOAD, 3);
+            mv.visitInsn(ARETURN);
+            mv.visitLabel(LEnd);
+
+            mv.visitTryCatchBlock(L0, L1, L2, "java/lang/NoSuchMethodException");
+            mv.visitTryCatchBlock(L0, L1, L2, "java/lang/ClassNotFoundException");
+            mv.visitTryCatchBlock(L0, L1, L2, "java/lang/reflect/InvocationTargetException");
+            mv.visitTryCatchBlock(L0, L1, L2, "java/lang/IllegalAccessException");
+
+            mv.visitLocalVariable("cache", "Lio/netty/buffer/PoolThreadCache;", null, testHashMap, LEnd, 3);
+            mv.visitLocalVariable("this", "Lio/netty/buffer/PooledByteBufAllocator;", null, LStart, LEnd, 0);
+            mv.visitLocalVariable("initialCapacity", "I", null, LStart, LEnd, 1);
+            mv.visitLocalVariable("maxCapacity", "I", null, LStart, LEnd, 2);
+            mv.visitLocalVariable("method", "Ljava/lang/reflect/Method;", null, L0, LEnd, 4);
+            mv.visitLocalVariable("currentCarrierThread", "Ljava/lang/Thread;", null, LStart, LEnd, 5);
+            mv.visitLocalVariable("e", "Ljava/lang/ReflectiveOperationException;", null,
+                    LError, testHashMap, 6);
+            mv.visitLocalVariable("lthreadCaches", "Ljava/util/concurrent/ConcurrentHashMap;",
+                    "Ljava/util/concurrent/ConcurrentHashMap<Ljava/lang/Thread;Lio/netty/buffer/PoolThreadCache;>;",
+                    testHashMap, LEnd, 7);
+            mv.visitLocalVariable("heapArena", "Lio/netty/buffer/PoolArena;",
+                    "Lio/netty/buffer/PoolArena<[B>;", LKeyOut, LEnd, 6);
+            mv.visitLocalVariable("directArena", "Lio/netty/buffer/PoolArena;",
+                    "Lio/netty/buffer/PoolArena<[B>;", LKeyOut, LEnd, 9);
+
+            mv.visitMaxs(5, 10);
+        }
+
+        @Override
+        public void visitEnd() {
+            cv.visitField(ACC_STATIC | ACC_PRIVATE, "isVirtualHandle",
```

Comment:
@anavarr `MethodHandle` to deliver the right level of performance should be `static final`!

Same applies to `threadCaches`: to be correctly published it should be `final` as well!